src/common/coap_libcoap.cpp

```diff
@@ -269,7 +269,7 @@ void AgentLibcoap::HandleResponse(coap_context_t *       aCoap,
 
     VerifyOrExit(aSent != NULL, otbrLog(OTBR_LOG_ERR, "request not found!"));
 
-    memcpy(&meta, aSent->hdr + aSent->length, sizeof(meta));
+    memcpy(&meta, (unsigned char *)aSent->hdr + aSent->length, sizeof(meta));
```

Comment:
nit: use `static_cast<uint8_t*>(pdu->hdr)`?

Reply:
I used (unsigned char *) because this is how it's implemented in libcoap in exactly the same cases.